cmd/kubeadm/app/util/arguments.go

```diff
@@ -24,41 +24,54 @@ import (
 	"github.com/pkg/errors"
 
 	"k8s.io/klog/v2"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 )
 
-// BuildArgumentListFromMap takes two string-string maps, one with the base arguments and one
+// ArgumentsToCommand takes two Arg slices, one with the base arguments and one
 // with optional override arguments. In the return list override arguments will precede base
-// arguments
-func BuildArgumentListFromMap(baseArguments map[string]string, overrideArguments map[string]string) []string {
+// arguments. If an argument is present in the overrides, it will cause
+// all instances of the same argument in the base list to be discarded, leaving
+// only the instances of this argument in the overrides to be applied.
+func ArgumentsToCommand(base []kubeadmapi.Arg, overrides []kubeadmapi.Arg) []string {
 	var command []string
-	var keys []string
-
-	argsMap := make(map[string]string)
-
-	for k, v := range baseArguments {
-		argsMap[k] = v
-	}
-
-	for k, v := range overrideArguments {
-		argsMap[k] = v
+	// Copy the base arguments into a new slice.
+	args := make([]kubeadmapi.Arg, len(base))
+	copy(args, base)
```

Comment:
i am trying to understand how that would be different. can you please provide some examples? the idea was to allow a single --foo flag to remove all instances of --foo in the base, so that extra args do a true override.

Reply:
this saves us some time of removing the element from the `base`.

```glang
args = append(args[:j], args[j+1:]...)
```

we still need to append after searching and comparison.

Reply:
as mentioned here:
https://github.com/kubernetes/kubernetes/pull/119156/files#r1264666750
you can send a follow up with the optimizations if you'd like.

Reply:
+1 for using a set to track overrides flags, and just copy the flags that are not overwritten.